cartography/intel/aws/permission_relationships.py

```diff
@@ -0,0 +1,338 @@
+    """
+    results = neo4j_session.run(
+        get_policy_query,
+        AccountId=account_id,
+    )
+    principals = {}
+    for r in results:
+        principal_arn = r["principal_arn"]
+        policy_id = r["policy_id"]
+        statements = r["statements"]
+        if principal_arn not in principals:
+            principals[principal_arn] = {}
+        principals[principal_arn][policy_id] = compile_statement(parse_statement_node(statements))
+    return principals
+
+
+def get_resource_arns(neo4j_session, account_id, node_label):
+    get_resource_query = Template("""
+    MATCH (acc:AWSAccount{id:{AccountId}})-[:RESOURCE]->(resource:$node_label)
+    return resource.arn as arn
+    """)
+    get_resource_query = get_resource_query.safe_substitute(node_label=node_label)
+    results = neo4j_session.run(
+        get_resource_query,
+        AccountId=account_id,
+    )
+    arns = [r["arn"] for r in results]
+    return arns
+
+
+def load_principal_mappings(neo4j_session, principal_mappings, node_label, relationship_name, update_tag):
+    map_policy_query = Template("""
+    UNWIND {Mapping} as mapping
+    MATCH (principal:AWSPrincipal{arn:mapping.principal_arn})
+    MATCH (resource:$node_label{arn:mapping.resource_arn})
+    MERGE (principal)-[r:$relationship_name]->(resource)
+    SET r.lastupdated = {aws_update_tag}
+    """)
+    if not principal_mappings:
+        return
+    map_policy_query = map_policy_query.safe_substitute(
+        node_label=node_label,
+        relationship_name=relationship_name,
+    )
+    neo4j_session.run(
+        map_policy_query,
+        Mapping=principal_mappings,
+        aws_update_tag=update_tag,
+    )
+
+
+def cleanup_rpr(neo4j_session, node_label, relationship_name, update_tag, current_aws_id):
+    logger.info("Cleaning up relationship '%s' for node label '%s'", relationship_name, node_label)
+    cleanup_rpr_query = Template("""
+        MATCH (:AWSAccount{id: {AWS_ID}})-[:RESOURCE]->(principal:AWSPrincipal)-[r:$relationship_name]->
+        (resource:$node_label)
+        WHERE r.lastupdated <> {UPDATE_TAG}
+        WITH r LIMIT {LIMIT_SIZE}  DELETE (r) return COUNT(*) as TotalCompleted
+    """)
+    cleanup_rpr_query = cleanup_rpr_query.safe_substitute(
+        node_label=node_label,
+        relationship_name=relationship_name,
+    )
+
+    statement = GraphStatement(cleanup_rpr_query, {'UPDATE_TAG': update_tag, 'AWS_ID': current_aws_id}, True, 1000)
+    statement.run(neo4j_session)
+
+
+def parse_permission_relationship_file(file):
+    try:
+        if not os.path.isabs(file):
+            file = os.path.join(os.getcwd(), file)
+        with open(file) as f:
+            relationship_mapping = yaml.load(f, Loader=yaml.FullLoader)
+        return relationship_mapping
+    except FileNotFoundError:
+        logger.warn(f"Permission relationshp mapping file {file} not found, skipping injestion ")
```

Comment:
Pycharm is complaining that we should use logger.warning instead because .warn is deprecated.

Reply:
fixed